pandas/tests/groupby/test_libgroupby.py

```diff
@@ -302,3 +303,20 @@ def test_cython_group_mean_Inf_at_begining_and_end():
         actual,
         expected,
     )
+
+
+def test_cython_group_sum_Inf_at_begining_and_end():
+    # GH #53606
+    actual = np.array([[np.nan], [np.nan]], dtype="float64")
+    counts = np.array([0, 0], dtype="int64")
+    data = np.array([[np.inf], [np.inf], [np.inf]], dtype="float64")
```

Comment:
Can you also test 

 - (-np.inf) + (-np.inf)
 - np.inf + (-np.inf)
 - (-np.inf) + np.inf

I'd recommend parametrizing the data & expected values